x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/ToCharTestScript.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.sql.expression.function.scalar.datetime;
+
+import com.carrotsearch.randomizedtesting.generators.RandomNumbers;
+import org.elasticsearch.common.SuppressForbidden;
+import org.elasticsearch.common.util.set.Sets;
+
+import java.math.BigDecimal;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static java.util.Arrays.asList;
+import static java.util.regex.Pattern.quote;
+import static org.elasticsearch.xpack.sql.expression.function.scalar.datetime.DateTimeTestUtils.dateTime;
+
+/**
+ * Generates an psql file that can be used to generate the test dataset for the {@link DateTimeToCharProcessorTests}.
+ */
+public class ToCharTestScript {
+
+    private static class TestRecord {
+
+        // timezones that are valid both in Java and in Postgres
+        public static final List<String> POSTGRES_TEST_ZONE_LIST = asList(
```

Comment:
My initial comment stands - why aren't these strings externalized as in declared into a separate file?
I'm not clear how they were produced nor why they have to be declared inside the java file and not separately?
I like the idea of having everything in one place however in this case the input clearly comes from postgres and since the input is actually external, why not apply that to the zone list as well considering the declaration is ~ 40 strings.

Reply:
The list of time zones here are a few hand-picked that are available in older JDK versions (supported, CI runs against it), latest JDK versions and also in Postgres. These are inputs to the generator and show up in the generated sql script that you can run against PostgreSQL to (re)generate the expected `TO_CHAR()` outputs.

What about if instead of burning these in, I put the output of the `SELECT name from pg_timezone_names` into a file and cross-check it against the Java timezone list? The one callout I have to add is that the `ToCharTestScript` will have to be run against the oldest supported JDK version, otherwise the tests on the CI will fail because of unsupported time zones.

Reply:
If there's a way to generate the list, even better - put the output into a file along a header that indicates how the list came to be and make the test read it.
I don't to over-engineer too much this PR which took significant time.

Reply:
Done.